From b11e8e294940615c844980ef1d49693c8442469a Mon Sep 17 00:00:00 2001
From: LLewellyn Fernandes <llewellyn.fernandes@evologics.de>
Date: Thu, 11 Jan 2024 14:42:37 +0100
Subject: [PATCH] remove dependency on qt5 and rauc

---
 conf/layer.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conf/layer.conf b/conf/layer.conf
index b59c8ea..694bdce 100644
--- a/conf/layer.conf
+++ b/conf/layer.conf
@@ -12,7 +12,7 @@ LICENSE_PATH += "${LAYERDIR}/licenses"
 CERT_PATH ??= "${OEROOT}/../../phytec-dev-ca"
 
 # Layer dependencies
-LAYERDEPENDS_yogurt = "phytec openembedded-layer networking-layer meta-python multimedia-layer qt5-layer rauc"
+LAYERDEPENDS_yogurt = "phytec openembedded-layer networking-layer meta-python multimedia-layer"
 LAYERSERIES_COMPAT_yogurt = "dunfell"
 
 # Let us add layer-specific bbappends which are only applied when that
-- 
2.25.1

